pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -503,13 +522,31 @@ func (c *ReverseExpandQuery) reverseExpandDirect(
 
 		tk := t.GetKey()
 
-		foundObject := tk.GetObject()
+		tupleCondition := tk.GetCondition()
+		conditionName := tupleCondition.GetName()
+		if conditionName != "" {
+			evaluableCondition := c.typesystem.GetCondition(conditionName)
+			conditionEvalResult, err := evaluableCondition.Evaluate(
+				req.context.AsMap(),
+				tupleCondition.GetContext().AsMap(),
+			)
```

Comment:
Can we use `TupleConditionMet` here, like we are doing inside `reverseExpandTupleToUserset`

Reply:
Definitely. That was just an oversight. All of these bits were meant to use `TupleConditionMet`.